tests/test_communication/mock_utransport.py

```diff
@@ -82,57 +85,74 @@ def unregister_listener(self, source: UUri, listener: UListener, sink: UUri = No
             result = UStatus(code=code)
         return result
 
-    def send(self, message: UMessage) -> UStatus:
+    async def send(self, message: UMessage) -> UStatus:
         validator = UAttributesValidator.get_validator(message.attributes)
         with self.lock:
             if message is None or validator.validate(message.attributes) != ValidationResult.success():
                 return UStatus(code=UCode.INVALID_ARGUMENT, message="Invalid message attributes")
 
-            executor = ThreadPoolExecutor(max_workers=5)
-            executor.submit(self._notify_listeners, message)
+            # Use a ThreadPoolExecutor with max_workers=1
+            executor = ThreadPoolExecutor(max_workers=1)
```

Comment:
Probably okay (especially since this is a test), but it's best IMO not to mix concurrent.futures with asyncio. This is the only place where you're using concurrent I think. Everywhere else is asyncio

Reply:
This is just a sample implementation and it will be modified in the next PR

Reply:
sounds good, no issues then